app/uk/gov/hmrc/exports/services/mapping/ExportsPointerToWCOPointer.scala

```diff
@@ -120,3 +105,61 @@ class ExportsPointerToWCOPointer @Inject() (environment: Environment) {
 
   private def isSequenceNumber(segment: String): Boolean = segment.startsWith("#") && segment.drop(1).toIntOption.nonEmpty
 }
+
+private case class Pointers(cds: String, wco: String, flags: Option[Seq[(String, String)]], comments: Option[Seq[String]])
+
+private object PointersReads extends ConstraintReads {
+
+  private val regex = "^(?!declaration\\.).+".r
+
+  val cdsReads: Reads[String] =
+    filter(JsonValidationError("Expecting non empty rows NOT starting with 'declaration.'")) { cds =>
+      regex.matches(cds) && cds.nonEmpty
+    }
+
+  val wcoReads: Reads[String] =
+    filter(JsonValidationError("Empty WCO pointer"))(_.nonEmpty)
+
+  val flagsReads: Reads[Seq[(String, String)]] = Reads[Seq[(String, String)]] { json =>
+    json.validate[Seq[JsObject]].map { objects =>
+      objects.flatMap(_.fields).map { case (key, value) =>
+        (key, value.as[String])
+      }
+    }
+  }
+
+  // The following method is a temporary hack to handle pointers that come from the frontend as a sequence, but for reasons of expediency we are mapping to the WCO Pointer that is actually a parent of the sequenced elements.
+  // Example: There can be up to 99 taric or nact codes, but we will point to the Commodity element if any of these change.
+  // These are handled by removing the final sequence id so that this class will parse them.
+  def convertManyToOnePointers(pointer: String): String =
+    if (Seq("items.$1.nactCode.$2", "items.$1.taricCode.$2", "items.$1.procedureCodes.additionalProcedureCodes.$2").contains(pointer))
+      pointer.dropRight(3)
+    else pointer
+
+}
+
+private object Pointers {
+
+  def apply(cds: String, wco: String, flags: Option[Seq[(String, String)]] = None, comments: Option[Seq[String]] = None): Pointers =
+    new Pointers(s"declaration.$cds".replace("$", ""), wco, flags, comments)
+
+  implicit val reads: Reads[Pointers] = new Reads[Pointers] {
+
+    import PointersReads._
+
+    override def reads(json: JsValue): JsResult[Pointers] = {
+
+      val cds: String = (json \ "cds").as[String](cdsReads)
+      val wco: String = (json \ "wco").as[String](wcoReads)
+
+      val flags: Option[Seq[(String, String)]] = (json \ "flags").asOpt[Seq[(String, String)]](flagsReads)
+      val comments: Option[Seq[String]] = (json \ "comments").asOpt[Seq[String]]
+
+      if (convertManyToOnePointers(cds).count(_ == '$') == wco.count(_ == '$')) {
```

Comment:
The pointers referred to in the body of `convertManyToOnePointers` need a flag indicating they need to be processed in a certain way (ie having the sequenceId and last full stop removed). So we'll be removing the method but keeping the logic.

Reply:
Although, the main reason I had this logic, which was to make the  number of "$" chars match between CDS and WCO pointers, is not here anymore so there isn't actually a need to do this.

Reply:
So the pointers mentioned in `convertManyToOnePointers` will need comments indicating they are not one to one pointers in the json file, which I'll add above.

Reply:
But this can be removed 158-160.

Reply:
And 131- 139